cache/cache.go

```diff
@@ -0,0 +1,129 @@
+package cache
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/caarlos0/env"
+	"github.com/gomodule/redigo/redis"
+	"github.com/pkg/errors"
+)
+
+type config struct {
+	Server      string        `env:"CACHE_SERVER" envDefault:"127.0.0.1:6379"`
+	MaxIdle     int           `env:"CACHE_MAX_IDLE" envDefault:"10"`
+	MaxActive   int           `env:"CACHE_MAX_ACTIVE" envDefault:"100"`
+	IdleTimeout time.Duration `env:"CACHE_IDLE_TIMEOUT" envDefault:"24s"`
+	Wait        bool          `env:"CACHE_WAIT" envDefault:"true"`
+}
+
+/*CachePool maintains a pool of connections.The application calls the Get method to get
+a connection from the pool and the connection's Close method to return the
+connection's resources to the pool.*/
+var CachePool *redis.Pool
+
+func init() {
+	cfg := config{}
+
+	if err := env.Parse(&cfg); err != nil {
+		fmt.Printf("%+v\n", err)
+	}
+
+	CachePool = &redis.Pool{
+		MaxIdle:     cfg.MaxIdle,
+		IdleTimeout: cfg.IdleTimeout,
+		MaxActive:   cfg.MaxActive,
+		Wait:        cfg.Wait,
+		Dial: func() (redis.Conn, error) {
+			c, err := redis.Dial("tcp", cfg.Server)
+			if err != nil {
+				log.Fatal(err)
+				return nil, err
+			}
+			return c, nil
+		},
+	}
+}
+
+//SetEx sets a new key in to database and also expiry time in ms
+func SetEx(RConn *redis.Conn, key string, ttl int, data interface{}) (interface{}, error) {
+	return (*RConn).Do("SETEX", key, ttl, data)
+}
+
+//GetString returns the value for the key in string format
+func GetString(RConn *redis.Conn, key string) (string, error) {
+	//get
+	return redis.String((*RConn).Do("GET", key))
```

Comment:
Doesn't this also do the same?
```suggestion
	return redis.String(RConn.Do("GET", key))
```

Reply:
I think they are different since  get this err `RConn.Do undefined (type *redis.Conn is pointer to interface, not interface)`